libs/wingsdk/src/shared-aws/function.ts

```diff
@@ -35,10 +34,6 @@ export class Function {
       return host;
     }
 
-    if (host instanceof AwsCdkFunction) {
```

Comment:
I think duck-typing much be a better approach here.

Basically check that the host implements `IAwsFunction`'s methods.